include/ginkgo/core/log/logger.hpp

```diff
@@ -594,7 +594,7 @@ class EnableLogging : public Loggable {
     void remove_logger(const Logger *logger) override
     {
         auto idx = find_if(begin(loggers_), end(loggers_),
-                           [&logger](std::shared_ptr<const Logger> l) {
+                           [&logger](std::shared_ptr<const Logger> &l) {
```

Comment:
tiny nit:
```suggestion
                           [&logger](const std::shared_ptr<const Logger> &l) {
```
or `const auto& l` would make sure we don't modify the smart pointer.